crates/core/src/operations/optimize.rs

```diff
@@ -331,27 +347,21 @@ impl<'a> std::future::IntoFuture for OptimizeBuilder<'a> {
             let operation_id = this.get_operation_id();
             this.pre_execute(operation_id).await?;
 
-            let writer_properties = this.writer_properties.unwrap_or_else(|| {
-                WriterProperties::builder()
-                    .set_compression(Compression::ZSTD(ZstdLevel::try_new(4).unwrap()))
-                    .set_created_by(format!("delta-rs version {}", crate_version()))
-                    .build()
-            });
```

Comment:
TODO.  Right now, we're breaking the existing behaviour of using zstd for compaction operations and this will use snappy by default. This is OK for getting feedback on the initial PR, but we should make sure to address this before merging.